vertx-sql-client/src/main/java/io/vertx/sqlclient/SqlConnectOptions.java

```diff
@@ -58,60 +55,96 @@ public SqlConnectOptions(JsonObject json) {
 
   public SqlConnectOptions(SqlConnectOptions other) {
     super(other);
-    this.host = other.host;
-    this.port = other.port;
+    if (other.hosts != null) {
+      setHosts(other.hosts);
+    }
     this.user = other.user;
     this.password = other.password;
     this.database = other.database;
     this.cachePreparedStatements = other.cachePreparedStatements;
     this.preparedStatementCacheMaxSize = other.preparedStatementCacheMaxSize;
     this.preparedStatementCacheSqlFilter = other.preparedStatementCacheSqlFilter;
+    this.tracingPolicy = other.tracingPolicy;
     if (other.properties != null) {
       this.properties = new HashMap<>(other.properties);
     }
   }
 
   /**
+   * @deprecated use {@link SqlConnectOptions#getHosts()}
+   *
    * Get the host for connecting to the server.
    *
    * @return the host
    */
+  @Deprecated
+  @GenIgnore
```

Comment:
Ignored to make json converters use `getHosts()`
Setters are not ignored due to backward compatibility reasons, i.e. to construct instance of options from json generated from old version.